### PR TITLE
throwIfPossible usage flaws fix

### PR DESCRIPTION
**Pull request by ntfshard**
I spotted several places where using throwIfPossible leads to unexpected behaviour if exceptions are: disabled(3 places, 2 files with +return); enabled(shall we release block in this case). 
But I'm not quite sure. Could you please help me figure out ;)

### CODE REVIEW

File: include/algorithms/multi_class_classifier/multi_class_classifier_model_builder.h [hunk 1]
```diff
@@ -82,11 +82,12 @@ class DAAL_EXPORT ModelBuilder
      */
     void setTwoClassClassifierModel(size_t negativeClassIdx, size_t positiveClassIdx, const classifier::ModelPtr& model)
     {
-        if(negativeClassIdx >= positiveClassIdx)
+        if(negativeClassIdx >= positiveClassIdx || !_s)
         {
             _s = services::Status(services::ErrorIncorrectParameter);
+            services::throwIfPossible(_s);
+            return;
```
Comment (ntfshard) [inline include/algorithms/multi_class_classifier/multi_class_classifier_model_builder.h]: no, that's no right

File: include/data_management/data_source/modifiers/csv/internal/default_modifiers.h [hunk 2]
```diff
@@ -168,6 +168,7 @@ class CategoricalFeatureModifier : public FeatureModifier
         if ( !_primitives.data() )
         {
             services::throwIfPossible(services::ErrorMemoryAllocationFailed);
+            return;
```
Comment (ShvetsKS) [inline include/data_management/data_source/modifiers/csv/internal/default_modifiers.h]: Looks like we can`t update status unfortunately.

File: include/algorithms/multi_class_classifier/multi_class_classifier_model_builder.h [hunk 3]
```diff
@@ -82,11 +82,12 @@ class DAAL_EXPORT ModelBuilder
      */
     void setTwoClassClassifierModel(size_t negativeClassIdx, size_t positiveClassIdx, const classifier::ModelPtr& model)
     {
-        if(negativeClassIdx >= positiveClassIdx)
+        if(negativeClassIdx >= positiveClassIdx || !_s)
```
Comment (ShvetsKS) [inline include/algorithms/multi_class_classifier/multi_class_classifier_model_builder.h]: if status is not ok
seems we have to update status with 'ErrorMemoryAllocationFailed'
we have to differentiate checks

**Comment by ntfshard** [comment]
make test

**Comment by ntfshard** [comment]
make test

**Pull request merged**